absinthe_plug.exs

```diff
@@ -0,0 +1,74 @@
+Mix.install([
+  {:absinthe, "~> 1.7"},
+  {:absinthe_plug, "~> 1.5"},
+  {:jason, "~> 1.4"},
+  {:plug_cowboy, "~> 2.5"},
+  {:bandit, "~> 1.1"}
+])
+
+defmodule ContentTypes do
+  use Absinthe.Schema.Notation
+
+  object(:post) do
+    field(:id, :id)
+    field(:title, :string)
+    field(:body, :string)
+  end
+end
+
+defmodule Schema do
+  use Absinthe.Schema
+
+  import_types(ContentTypes)
+
+  query do
+    field :posts, list_of(:post) do
+      resolve(fn _parent, _args, _context ->
+        posts = [
+          %{
+            id: 1,
+            title: "Foo",
+            body: "Bar"
+          }
+        ]
+
+        {:ok, posts}
+      end)
+    end
+  end
+end
+
+defmodule Router do
+  use Plug.Router
+  plug(Plug.Logger)
+  plug(:match)
+  plug(:dispatch)
+
+  plug(Plug.Parsers,
+    parsers: [:urlencoded, :multipart, :json, Absinthe.Plug.Parser],
+    pass: ["*/*"],
+    json_decoder: Jason
+  )
+
+  forward("/api",
+    to: Absinthe.Plug,
+    init_opts: [schema: Schema]
+  )
+end
+
+plug_cowboy = {Plug.Cowboy, plug: Router, scheme: :http, port: 4000}
+bandit = {Bandit, plug: Router, scheme: :http, port: 4001}
+require Logger
+Logger.info("starting #{inspect(plug_cowboy)}")
+{:ok, _} = Supervisor.start_link([plug_cowboy], strategy: :one_for_one)
+{:ok, _} = Supervisor.start_link([bandit], strategy: :one_for_one)
```

Comment:
why do we have two servers? Just to showcase how to use with cowboy or bandit? If so, let's keep just bandit, it's very easy to switch around.

Reply:
done!